src/types.rs

```diff
@@ -15,6 +15,14 @@ pub struct VersionData {
     pub version: String,
 }
 
+#[derive(Serialize, Deserialize, Debug)]
+pub struct Coordinate {
+    #[serde(with = "crate::serde::as_string")]
+    slot: Slot,
+    root: Root,
+    execution_optimistic: bool,
+}
+
```

Comment:
if you wanted to do something like this: https://github.com/ralexstokes/beacon-api-client/blob/main/src/types.rs#L357

we could still reuse the `Coordinate` type from `ethereum-consensus` and flatten the other bits into a map

so you could try something like:

```rust
pub struct CoordinateWithMetadata {
    #[serde(flatten)]
    pub coordinate: Coordinate,
    #[serde(flatten)]
    pub meta: HashMap<String, serde_json::Value>,
}
```

it is minor here bc we only have the few fields but I do like transparently getting updates with a single definition vs many we have to manually synchronize

Reply:
Ok, except there was an additional problem parsing the returned value directly into `slot` as type `Slot` for the imported `Coordinate`, hence the additional serde wrapper to wrangle it from `String` in the explicitly defined version.

Reply:
ah, yeah you could add the `serde::as_string` bit to the `Slot` there in a PR if you want

I think its ok to duplicate this here if you want

Reply:
can try this now w/ https://github.com/ralexstokes/ethereum-consensus/pull/161

will need to update the dependency rev in the `Cargo.toml` too